contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -297,6 +299,34 @@ func (h *htlcIncomingContestResolver) Resolve() (ContractResolver, error) {
 		}
 
 		witnessUpdates = preimageSubscription.WitnessUpdates
+
+		// Also subscribe to the htlc interceptor in case a settle comes
+		// in from that direction.
+		interceptorUpdates = make(chan lntypes.Preimage, 1)
+
+		packet := &InterceptedPacket{
+			Hash:           h.htlc.RHash,
+			IncomingExpiry: h.htlcExpiry,
+			IncomingAmount: h.htlc.Amt,
+			IncomingCircuit: channeldb.CircuitKey{
+				ChanID: h.htlcSuccessResolver.ShortChanID,
+				HtlcID: h.htlc.HtlcIndex,
+			},
+			OutgoingChanID: payload.FwdInfo.NextHop,
+			OutgoingExpiry: payload.FwdInfo.OutgoingCTLV,
+			OutgoingAmount: payload.FwdInfo.AmountToForward,
+			CustomRecords:  payload.CustomRecords(),
+		}
+		copy(packet.OnionBlob[:], nextHopOnionBlob)
+
+		fwd := newInterceptedForward(packet, interceptorUpdates)
+
+		// Ignore intercepted boolean. We only need to listen on the
+		// interceptorUpdates channel.
+		err := h.HtlcInterceptor(fwd)
```

Comment:
it should be reasonable to sassume that the packet is already being intercepted at this point, right? If so, then https://github.com/lightningnetwork/lnd/pull/6219/commits/b18362425e521fcb3e0b8f9287c3807756b14387#diff-738e9811e26d363a1a7e6ea2b36b89f3af0f054ca550d94f83a60faab4122fd8R351 will make it so that `interceptorUpdates` will never get notified.

Reply:
There could have been a restart in between.

Also, it might be that the interceptor already signaled settle and the link went down before settle was locked-in. In that case, the onchain resolver will offer the packet to the interceptor again.